app/src/main/java/dev/shreyaspatil/foodium/ui/details/PostDetailsActivity.kt

```diff
@@ -109,6 +112,12 @@ class PostDetailsActivity : BaseActivity<PostDetailsViewModel, ActivityPostDetai
     }
 
     companion object {
-        const val POST_ID = "postId"
+        private const val POST_ID = "postId"
+
+        fun start(context: Context, postId: Int?, options: ActivityOptionsCompat) {
+            val intent = Intent(context, PostDetailsActivity::class.java)
+            intent.putExtra(POST_ID, postId)
+            context.startActivity(intent, options.toBundle())
+        }
```

Comment:
This function shouldn't start activity internally using `Context` of the other `Activity`. I referred that blog link which you shared. Means we should be confident who's starting the activity. So it would be great if we start the activity from MainActivity itself. 

I would suggest to make this method like... 👇🏻 

```kotlin

fun getStartIntent(context: Context, postId: Int) = Intent(context, PostDetailsActivity::class.java).apply {
    putExtra(POST_ID, postId)
}
```

From MainActivity, let's do like... 👇🏻 

```kotlin
val intent = PostDetailsActivity.getStartIntent(this, post.id)
startActivity(intent, options.toBundle())
```

Reply:
That makes sense. Thanks for the quick review and clarification. I will make the necessary changes and push it. 😄